### PR TITLE
Onboarding Screen

### PR DESCRIPTION
**Pull request by adamstener**
This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [x] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [x] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [x] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: secant/UI Components/CircularFrame/CircularFrameBadge.swift [hunk 1]
```diff
@@ -6,23 +6,70 @@
...
+                                Spacer()
 
-    let badge: Badge
+                                ZStack {
+                                    ForEach(0..<viewStore.badges.count) { badgeIndex in
+                                        viewStore.badges[viewStore.index].image
+                                            .resizable()
+                                            .renderingMode(.none)
+                                            .frame(
+                                                width: proxy.size.width * 0.5,
```
[the remark this answers is not in the document]
Comment (adamstener) [reply secant/UI Components/CircularFrame/CircularFrameBadge.swift:50]: We can certainly pull out any values into constants so they have names, but I'm not certain that would solve anything here in this example.  Happy to take suggestions though if you have ideas!

[the remark this answers is not in the document]
Comment (pacu) [reply secant/UI Components/CircularFrame/CircularFrameBadge.swift:50]: yeah! sometimes, the number it's just unavoidable and naming it just creates yet another thing to maintain. 
The most epic fail thing I saw in code  along these lines was something like this:
```` Swift
let twentySixPixelHeightButton: CGFloat = 27
```` 

you can imagine how the rest of the code was. 😂

[the remark this answers is not in the document]
Comment (adamstener) [reply secant/UI Components/CircularFrame/CircularFrameBadge.swift:50]: This is almost a good example of something good though.  If for example you did have a button you wanted to have a constant height of 27, some people are ok with just hard coding that value if its only used once, or you could pull it out into a property named `buttonHeight` or something similar.  I'm certainly against magic numbers that are repeated, or used multiple times, and also am right there with you that sometimes adding context with a property name does help the reader.

**Review by pacu — APPROVED** [review]

**Comment by pacu** [comment]
This looks pretty awesome! Great job @adamstener

File: secant.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1126,7 +1160,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 3;
 				DEVELOPMENT_ASSET_PATHS = "\"secant/Preview Content\"";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = LNXNCURQGQ;
```
Comment (dh-ecc) [inline secant.xcodeproj/project.pbxproj:1163]: [Question / maybe nit]
Is this intentional? (if not let's remove 😊)

File: secant.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1150,7 +1184,7 @@
...
```
Comment (dh-ecc) [inline secant.xcodeproj/project.pbxproj:1187]: [Question / maybe nit]
Is this intentional? (if not let's remove 😊)

File: secant/Features/Onboarding/OnboardingStore.swift [hunk 4]
```diff
@@ -9,20 +9,22 @@ import Foundation
...
+    var steps: IdentifiedArrayOf<Step> = Self.onboardingSteps
     var index = 0
     var skippedAtindex: Int?
     
-    var currentStep: OnboardingStep { steps[index] }
-    var skipButtonDisabled: Bool { steps.count == index + 1 }
-    var backButtonDisabled: Bool { index == 0 }
+    var currentStep: Step { steps[index] }
+    var finalStep: Bool { steps.count == index + 1 }
+    var initialStep: Bool { index == 0 }
```
Comment (dh-ecc) [inline secant/Features/Onboarding/OnboardingStore.swift:27]: [non-blocking | naming | suggestion]

What do you think about prepending these properties with "is"?

```suggestion
    var isFinalStep: Bool { steps.count == index + 1 }
    var isInitialStep: Bool { index == 0 }
```

**Review by dh-ecc — COMMENTED** [review]
This looks great!

Put in a couple low impact inline comments and have some "general" things below:

[Non-blocking | Question]
What was the criteria for putting views in the `Screens` directory vs inside the `Feature`'s directory?

**Comment by dh-ecc** [comment]
Oh also thanks for fixing the lint warnings I introduced 😅. 🙏

**Review by pacu — APPROVED** [review]

**Comment by pacu** [comment]
This Looks great.

**Review by Daniel1of1 — APPROVED** [review]
🚀

**Pull request merged**